### PR TITLE
Add support for a IBMCLOUD_CONFIG_HOME which is the absolute path to the config dir

### PR DESCRIPTION
**Pull request by rws-github**
In preparation to align IBM Cloud CLI and private cloud CLI.

### CODE REVIEW

File: bluemix/configuration/config_helpers/helpers_test.go [hunk 1]
```diff
@@ -10,6 +20,59 @@ package config_helpers
...
+	// if IBMCLOUD_CONFIG_HOME is set but does not exist, IBMCLOUD_HOME is used
+	os.Setenv("IBMCLOUD_CONFIG_HOME", "/my_ibmcloud_config_home")
+	assert.Equal(filepath.Join("/my_ibmcloud_home", ".bluemix"), ConfigDir())
+
+	// if IBMCLOUD_CONFIG_HOME is set and exists, IBMCLOUD_CONFIG_HOME is used
+	os.Setenv("IBMCLOUD_CONFIG_HOME", userHome)
+	assert.Equal(userHome, ConfigDir())
+}
+
 // func TestMigrateFromOldConfig(t *testing.T) {
```
Comment (weieigao) [inline bluemix/configuration/config_helpers/helpers_test.go]: The method TestConfigDir() covers many cases, better to put each in a separate test method.

File: bluemix/configuration/config_helpers/helpers_test.go [hunk 2]
```diff
@@ -10,6 +20,59 @@ package config_helpers
...
+	assert.Equal(filepath.Join("/my_bluemix_home", ".bluemix"), ConfigDir())
+
+	// if BLUEMIX_HOME and IBMCLOUD_HOME is set, IBMCLOUD_HOME is used
+	os.Setenv("IBMCLOUD_HOME", "/my_ibmcloud_home")
+	assert.Equal(filepath.Join("/my_ibmcloud_home", ".bluemix"), ConfigDir())
+
+	// if IBMCLOUD_CONFIG_HOME is set but does not exist, IBMCLOUD_HOME is used
+	os.Setenv("IBMCLOUD_CONFIG_HOME", "/my_ibmcloud_config_home")
+	assert.Equal(filepath.Join("/my_ibmcloud_home", ".bluemix"), ConfigDir())
+
```
Comment (weieigao) [inline bluemix/configuration/config_helpers/helpers_test.go]: Even if the user set the config dir via `IBMCLOUD_CONFIG_HOME`, SDK still pickup another config dir if `IBMCLOUD_CONFIG_HOME` not exist. I think it's a little confusing for user...
It's better to create the config dir if it not exist.

File: bluemix/configuration/config_helpers/helpers_test.go (hunk 1 above)
Comment (rws-github) [reply bluemix/configuration/config_helpers/helpers_test.go]: Test cases separated.

File: bluemix/configuration/config_helpers/helpers_test.go (hunk 2 above)
Comment (rws-github) [reply bluemix/configuration/config_helpers/helpers_test.go]: Changed the behavior to create the IBMCLOUD_CONFIG_HOME directory if it does not exist. Added tests to verify.

**Comment by rws-github** [comment]
@weieigao I have updated the pull request. Do you have any further concerns that I can address?

**Comment by weieigao** [comment]
@rws-github  I made a little change, see https://github.com/IBM-Cloud/ibm-cloud-cli-sdk/commit/431acd9809a2c5a1a6da37b624984957fcf3617a

You don't need to create the dir in advance, it will be created when used, see https://github.com/IBM-Cloud/ibm-cloud-cli-sdk/blob/master/bluemix/configuration/persistor.go#L58

**Comment by weieigao** [comment]
If you are ok with it, I will do the merge

**Comment by rws-github** [comment]
Yes, looks fine to me.

**Comment by weieigao** [comment]
merged https://github.com/IBM-Cloud/ibm-cloud-cli-sdk/commit/0cedd30163583d004bc839dfbeb737436e25134f

**Pull request closed without merging**